hack/prow/run_tests.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+# Copyright 2019 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script will execute a set of named minikube tests,
+   gather the results, logs, and artifacts into a named GCS
+   bucket for presentation in k8s testgrid:
+   https://k8s-testgrid.appspot.com
+"""
+
+import os, sys, json, re, argparse, calendar, time, subprocess, shlex
+
+def write_results(outdir, started, finished, test_results):
+  """ write current results into artifacts/junit_runner.xml 
+      format:
+      <testsuite failures="XXX" tests="YYY" time="ZZZ">
+         <testcase classname="SUITENAME" name="TESTNAME" time="ZZZ1" />
+         <testcase classname="SUITENAME" name="TESTNAME" time="ZZZ1" />
+         ...
+      </testsuite>
+
+      write the started.json and finish.json files 
+      format:
+      started.json: {"timestamp":STARTTIMEINSECONDSINCEEPOCH}
+      finished.json: {"timestamp":FINISHTIMEINSECONDSINCEEPOCH,
+                      "passed":FINALRESULT,
+                      "result":SUCCESS|FAIL,
+                      "metadata":{}
+                      }
+  """
+  started_json_file = open('%s/started.json' % outdir, 'w')
+  finished_json_file = open('%s/finished.json' % outdir, 'w')
+  junit_xml_file = open('%s/artifacts/junit_runner.xml' % outdir, 'w')
+
+  failures = 0
+  testxml = ""
+  for test in test_results:
+    testxml += '<testcase classname="%s" name="%s" time="%s">' % (test['classname'], test['name'], test['time'])
+    if test['status'] == 'FAIL':
+        failures += 1
+        testxml += '<failure message="Test Failed" />'
+    testxml += '</testcase>\n'
+  junit_xml_file.write('<testsuite failures="%s" tests="%s">\n' % (failures, len(test_results)))
+  junit_xml_file.write(testxml)
+  junit_xml_file.write('</testsuite>')
+  junit_xml_file.close()
+
+  started_json_file.write(json.dumps(started)) 
+  started_json_file.close()
+  finished_json_file.write(json.dumps(finished)) 
+  finished_json_file.close()
+
+  return
+
+def upload_results(outdir, test, buildnum, bucket):
+  """ push the contents of gcs_out/* into bucket/test/logs/buildnum"""
+  classname = os.path.basename(test).split('.')[0]
+  args = shlex.split("gsutil cp -R gcs_out/ gs://%s/logs/%s/%s" % (bucket, classname, buildnum))
```

Comment:
This should respect outdir.

Also, I think it'd be cleaner if this  just used a string array instead of using shlex.split to convert to one.